### PR TITLE
feat(condo): DOMA-10208 get payments by receipt

### PR DESCRIPTION
**Pull request by YEgorLu**
If user did not pay for virtual receipt or pay before receipt was made, we have Payment connected to Receipt by ID. We have function "getPaymentsSum", that searches for payments for receipt and calculates amount to pay. It was extended to search payments by receiptId, if it is present. So we want potentially remove searching by bank info of organization and ease paid amount calculation. Later we want to connect receipts with payments if payment was early or virtual and remove searching by bank info entirely.

### CODE REVIEW

File: apps/condo/domains/billing/utils/serverSchema/index.js [hunk 1]
```diff
@@ -69,15 +70,39 @@ async function registerBillingReceipts (context, data) {
...
-        accountNumber: accountNumber,
-        period: period,
-        status_in: [PAYMENT_DONE_STATUS, PAYMENT_WITHDRAWN_STATUS],
-        recipientBic: bic,
-        recipientBankAccount: bankAccount,
-    })
+const getPaymentsSum = async (context, receiptId, organizationId, accountNumber, period, bic, bankAccount) => {
+    const directReceiptIdWhere = { AND: [
+        { receipt: { id: receiptId } },
+        { receipt_is_null: false },
```
Comment (dkoviazin) [inline apps/condo/domains/billing/utils/serverSchema/index.js:76]: You do not need to check receipt_is_null here as it is already receipt: { id: receiptId }

File: apps/condo/domains/billing/utils/serverSchema/index.js [hunk 2]
```diff
@@ -69,15 +70,39 @@ async function registerBillingReceipts (context, data) {
...
+    ],
+    }
+
+    // TODO: create task to connect receipt to early payment after receipt creation
+    // TODO: remove this check after upper todo
```
Comment (dkoviazin) [inline apps/condo/domains/billing/utils/serverSchema/index.js:81]: you need to add task here

format is: // TODO(YEgorLu): DOMA-NNNN what you want to do

**Review by dkoviazin — APPROVED** [review]

**Review by abshnko — APPROVED** [review]

File: apps/condo/domains/billing/utils/serverSchema/index.js [hunk 3]
```diff
@@ -69,15 +70,37 @@ async function registerBillingReceipts (context, data) {
...
-        recipientBankAccount: bankAccount,
-    })
+const getPaymentsSum = async (context, receiptId, organizationId, accountNumber, period, bic, bankAccount) => {
+    const directReceiptIdWhere = { AND: [
+        { receipt: { id: receiptId } },
+    ],
+    }
+
+    // TODO(YEgorLu): DOMA-10222 create task to connect receipt to early payment after receipt creation and remove this check
+    const oldWhereByOrganizationCredentials = { AND: [
```
Comment (pahaz) [inline apps/condo/domains/billing/utils/serverSchema/index.js:80]: it looks too complex

File: apps/condo/domains/billing/schema/AllResidentBillingReceiptsService.test.js [hunk 4]
```diff
@@ -348,6 +348,47 @@ describe('AllResidentBillingReceiptsService', () => {
             expect(Big(receiptAfterPayment.paid).toFixed(2)).toEqual(paidAmounts.reduce((sum, amount) => Big(sum).add(Big(amount)).toFixed(2), '0.00'))
             expect(receiptAfterPayment.isPayable).toBeTruthy()
         })
+
+        test('paid field calculated when organization bank info changed', async () => {
```
Comment (pahaz) [inline apps/condo/domains/billing/schema/AllResidentBillingReceiptsService.test.js:352]: One test is not enought for this changes. You should test at least few negative cases.

File: apps/condo/domains/billing/utils/serverSchema/index.js (hunk 3 above)
Comment (YEgorLu) [reply apps/condo/domains/billing/utils/serverSchema/index.js:80]: it is...

This check almost duplicates old one. It is fields to know, that payment was made from certain user account to certain organization account so we can assume that payment was made for certain receipt and adds check for receiptId = null, to not to interfere with new logic

Will be removed from here after DOMA-10222 (most likely will live in new worker)

File: apps/condo/domains/billing/utils/serverSchema/index.js [hunk 5]
```diff
@@ -69,15 +70,37 @@ async function registerBillingReceipts (context, data) {
...
+        { organization: { id: organizationId } },
+        { accountNumber },
+        { period },
+        { receipt_is_null: true },
```
Comment (dkoviazin) [inline apps/condo/domains/billing/utils/serverSchema/index.js:84]: Do not forget to check that invoice is also null

**Comment by YEgorLu** [comment]
According to conversation, now searching payments only by connection to receipt.

File: apps/condo/domains/billing/schema/AllResidentBillingReceiptsService.test.js (hunk 4 above)
Comment (YEgorLu) [reply apps/condo/domains/billing/schema/AllResidentBillingReceiptsService.test.js:352]: Changed most tests to use payment creation with connection to receipt, and 1 to say that we do not count payments for virtual receipts at all

**Pull request merged**